### ISSUE TITLE
Custom sass importer

### ISSUE THREAD

**Issue by bencergazda** [skovy/typed-scss-modules#129]
## Expected Behavior

It looks like currently there's no possibility to provide importer functions for the sass compiler. This could be a huge benefit eg. when we try to `@use` a sass module from the node_modules folder, using `~` in the @use, which is a default behavior in weback's `sass-loader`. Unfortunately this breaks the type generation on the same file until we can pass an importer .

## Current Behavior

```scss
@use '~my-awesome-plugin/mixins';
```
This brakes the type generation, as sass cannot resolve `~` by default.

**Comment by skovy** [comment]
@bencergazda did you try passing the alias prefix? `--aliasPrefixes.~ node_modules/`

**Comment by bencergazda** [comment]
@skovy Unfortunately that works only if the SCSS file gets placed in the same folder as the `node_modules`.

It would be really great to be able to add extra options to the Sass compiler.

**Comment by bencergazda** [comment]
What about a `tsm.config.js` file it the cwd (I usually use cosmiconfig), with all the available CLI options and also add a `compilerOptions` property to MainOptions (or to SassOptions)?

**Comment by skovy** [comment]
@bencergazda could you create an example? I'm not fully following the use case and problem.

**Comment by bencergazda** [comment]
@skovy It happens quite often that we need to customize the Sass compiler's options, to add a custom Sass importer, or inject Sass functions etc. If this is the case, I need to be able to pass the same options to `typed-scss-modules`'s compiler, otherwise, the type generation will break.

(This is not something we could pass from the CLI, we need a config file to make this work.)

_Since reporting the issue, we wrote a webpack plugin with `typed-scss-modules` under the hood, that fits our needs, so we are already not in a need of this feature. If you feel like this is out of the project's scope, feel free to close the ticket._

**Comment by skovy** [comment]
@bencergazda got it, so it sounds like you need new options exposed here when the sass compiler is invoked? https://github.com/skovy/typed-scss-modules/blob/1391aa722eb309c76df6fdc9ce01d2a768984ff2/lib/sass/file-to-class-names.ts#L71-L73

Or more specifically, you can't really on a static string mapping and need to pass a JS function?

**Comment by mlp73** [comment]
Hi!
I am adding a +1 to that issue as my team is in the same situation. We are using a sass-loader with a custom importer to be able to import .json designtokens. The specific plugin is: node-sass-json-importer.
As we are not able to pass an additional importer to typed-scss-modules it fails when trying to resolve the import of our tokens.

@skovy any plan on opening up for additional importers?

**Comment by bencergazda** [comment]
> @bencergazda got it, so it sounds like you need new options exposed here when the sass compiler is invoked?
> 
> https://github.com/skovy/typed-scss-modules/blob/1391aa722eb309c76df6fdc9ce01d2a768984ff2/lib/sass/file-to-class-names.ts#L71-L73
> 
> Or more specifically, you can't really on a static string mapping and need to pass a JS function?

Yes, I would recommend adding the possibility to define a custom sass config object in a separate js file. That object could be merged with the project's own sass config (file, inlcudePaths, importer) in the renderSync call.

**Comment by skovy** [comment]
@bencergazda @mlp73 assuming both of you still use this package and need this functionality, could you please give `v5.1.1` a shot to test the behavior and verify it solves your use cases? See the linked PR for more details (or the updated README) but you should now be able to do the following after upgrading:

```js
// typed-scss-modules.config.js

const jsonImporter = require("node-sass-json-importer");

export const config = {
  importer: jsonImporter(),

  // all other CLI options...
  banner: "// config file banner",
  nameFormat: "kebab",
  exportType: "default",
  // ...
};

```

**Comment by byverdu** [comment]
It took me some time to figure out how to convert the CLI syntax `--aliasPrefixes.~ node_modules/` to config file one, just in case someone has the same issue it's like this

```js
export const config = {
   aliasPrefixes: {'~': 'node_modules/'},
}
```
